### PR TITLE
Bug

### PR DESCRIPTION
**Pull request by abdulkerimcan**
- changed project folder structure,
- added ApiConstants class as singleton variable
- consolidated three similar helper methods into a single method, eliminating code duplication
- changed unnecessary enums to extension to avoid memory leak
- applied dependency injection for the viewModel and service class

### CODE REVIEW

File: RickAndMortyApp/Views/DetailsCharacterCollectionViewCell.swift [hunk 1]
```diff
@@ -57,7 +57,7 @@ final class DetailsCharacterCollectionViewCell: UICollectionViewCell {
         }
     }
     
-    func setCell(viewModel: DetailsCharacterCellViewModel) {
+    func setCell(viewModel: LocationDetailsCharacterCellViewModel) {
         viewModel.registerForData { [weak self] data in
                     self?.imageView.af.setImage(withURL: URL(string: data.image) ?? URL(string: "https://rickandmortyapi.com/api/character/avatar/1.jpeg")!,
```
Comment (Scankiran) [inline RickAndMortyApp/Views/DetailsCharacterCollectionViewCell.swift]: You can use `URL(string: "https://rickandmortyapi.com/api/character/avatar/1.jpeg")!` as a constant. You are using some other cells and you can also use constant there.

File: RickAndMortyApp/Controllers/Core/TabbarVC.swift
Comment (Scankiran) [inline RickAndMortyApp/Controllers/Core/TabbarVC.swift:1]: You can create create UINavigationController to create functions. Also, can set tabbar item into functions. 
Let me add a suggestion here.

File: RickAndMortyApp/Controllers/Core/TabbarVC.swift [hunk 2]
```diff
@@ -44,7 +35,21 @@ final class TabbarVC: UITabBarController {
         }
         
         setViewControllers(navs, animated: true)
-        
+    }
+}
+
+private extension UIViewController {
+    func createCharacterVC() -> CharacterVC {
+        CharacterVC(viewModel: CharacterViewModel(service: Service()))
```
Comment (Scankiran) [inline RickAndMortyApp/Controllers/Core/TabbarVC.swift]:
```suggestion
        let characterVC = CharacterVC(viewModel: CharacterViewModel(service: Service()))
        let nav1 = UINavigationController(rootViewController: characterVC))
        nav1.tabBarItem = UITabBarItem(title: "Character", image: UIImage(systemName: "person"), tag: 1)
        nav1.navigationBar.prefersLargeTitles = true
        return nav1
```

File: RickAndMortyApp/Controllers/Core/TabbarVC.swift
Comment (Scankiran) [reply RickAndMortyApp/Controllers/Core/TabbarVC.swift:1]: Also you can delete for loop with suggestion block code.

File: RickAndMortyApp/Controllers/Others/CharacterDetailsScreen/CharacterDetailsViewModel.swift
Comment (Scankiran) [inline RickAndMortyApp/Controllers/Others/CharacterDetailsScreen/CharacterDetailsViewModel.swift:1]: Check 32. line here. There are idention issue. Try 'control+I'

File: RickAndMortyApp/Utilities/Constants/ApiConstants.swift
Comment (Scankiran) [inline RickAndMortyApp/Utilities/Constants/ApiConstants.swift:1]: add space lines between definitions.

**Comment by Scankiran** [comment]
Absolutely, a lot better. LGTM!

File: RickAndMortyApp/Views/DetailsCharacterCollectionViewCell.swift (hunk 1 above)
Comment (abdulkerimcan) [reply RickAndMortyApp/Views/DetailsCharacterCollectionViewCell.swift]: Done!

File: RickAndMortyApp/Controllers/Others/CharacterDetailsScreen/CharacterDetailsViewModel.swift
Comment (abdulkerimcan) [reply RickAndMortyApp/Controllers/Others/CharacterDetailsScreen/CharacterDetailsViewModel.swift:1]: Done!

File: RickAndMortyApp/Utilities/Constants/ApiConstants.swift
Comment (abdulkerimcan) [reply RickAndMortyApp/Utilities/Constants/ApiConstants.swift:1]: Done!

File: RickAndMortyApp/Controllers/Core/TabbarVC.swift
Comment (abdulkerimcan) [reply RickAndMortyApp/Controllers/Core/TabbarVC.swift:1]: Done!

**Comment by Scankiran** [comment]
LGTM! Can merge :)

**Pull request merged**